nautobot/extras/models/models.py

```diff
@@ -681,3 +682,26 @@ def log(
             else:
                 log_level = logging.INFO
             logger.log(log_level, str(message))
+
+
+@extras_features("graphql")
+class GraphqlQuery(BaseModel, ChangeLoggedModel):
+    name = models.CharField(max_length=255, unique=True)
+    slug = models.CharField(max_length=255, unique=True)
+    query = models.TextField()
+    variables = models.JSONField(default=dict)
+
+    class Meta:
+        ordering = ("slug",)
+
+    def get_absolute_url(self):
+        return reverse("extras:graphqlquery", kwargs={"slug": self.slug})
+
+    def save(self, *args, **kwargs):
+        query_vars = re.findall(r"\$[a-zA-Z0-9]+", self.query)
+        vars_object = {}
+        for var in query_vars:
+            vars_object[var[1:]] = ""
+
+        self.variables = json.dumps(vars_object)
```

Comment:
On the detailed view, i push out a JSON which can be filled in for testing the query. Having it as an actual JSON means it can be displayed without any modifications.
![Screenshot 2021-03-10 at 13 56 17](https://user-images.githubusercontent.com/6088317/110640170-6db50200-81a8-11eb-9d5c-479d5f6ee5ae.png)

Reply:
I also think it would be better to store the mandator variable as a list and let the use fill the proper variables for testing in  GraphiQL

Reply:
It also needs to be in JSON format to render the URLs to get passed into GraphiQL. If it is in list format I will need to add logic to the templates to convert the list into the dictionary

Reply:
```
{% for query in saved_graphiql_queries %}
                                <li>
                                    <a href="#" onclick="reload('{{ query.query|iriencode }}', '{{ query.variables|iriencode }}')">{{ query.name }}</a>
                                </li>
                            {% endfor %}
```
The variables need to be passed into the url as the dictionary here. I think it makes more sense to keep it as a dictionary with empty values so less processing needs to be done inside of the template.